Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -253,7 +253,7 @@ private void forceFileAnnotation(String file) {
     }
   }
 
-  public void addCompletions(@NotNull final String completionId, @NotNull final CompletionSuggestionProcessor processor) {
+  public void addCompletions(@NotNull final String completionId, @NotNull final Processor<CompletionSuggestion> processor) {
```

Comment:
yep, Processor is better. Consumer is even better when you never need to stop processing in the middle. By convention boolean returned by process() means "process further" whereas false means stop processing.